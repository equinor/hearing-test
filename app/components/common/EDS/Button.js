import React, { Component } from 'react';
import { Text, TouchableOpacity, View } from 'react-native';

export default class ButtonEDS extends Component<{ onPress: () => void, text: string }> {
  render() {
    return (
      <TouchableOpacity onPress={this.props.onPress}>
        <View
          style={{
            height: 62,
            backgroundColor: '#007079',
            alignItems: 'center',
            justifyContent: 'center',
            borderRadius: 4,
            marginBottom: 23,
          }}
        >
          <Text style={{ color: 'white', fontSize: 17 }}>{this.props.text}</Text>
        </View>
      </TouchableOpacity>
    );
  }
}
