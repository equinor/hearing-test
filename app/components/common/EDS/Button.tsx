import React, { Component } from 'react';
import { TouchableOpacity, View } from 'react-native';
import Spinner from '../atoms/Spinner';
import Typography from '../atoms/Typography';

export default class ButtonEDS extends Component<{
  onPress: () => void,
  text: string,
  outlined: boolean,
  small: boolean,
  danger: boolean,
  loading: boolean,
  disabled: boolean,
}> {
  getBackgroundColor(props) {
    if (props.disabled || props.loading || props.outlined) return `#EAEAEA`;
    if (props.danger) return '#EB0000';
    return '#007079';
  }

  render() {
    const { danger, disabled, loading, onPress, outlined, small, text } = this.props;
    return (
      <TouchableOpacity onPress={onPress} disabled={loading || disabled}>
        <View
          style={{
            height: 40,
            width:160,
            alignSelf: 'center',
            alignItems: 'center',
            justifyContent: 'center',
            borderRadius: 4,
            marginBottom: 18,
            borderWidth: !danger && 1,
            //width: small ? '50%' : '100%',
            backgroundColor: this.getBackgroundColor(this.props),
            borderColor: disabled || loading ? `#DCDCDC` : '#007079',
          }}
        >
          {loading ? (
            <Spinner color="#6F6F6F" />
          ) : (
            <Typography variant="button"
              color={outlined ? '#007079' : 'white'}
              style={{
                textAlign: 'center',
              }}
            >
              {text}
            
            </Typography>
          )}
        </View>
      </TouchableOpacity>
    );
  }
}
