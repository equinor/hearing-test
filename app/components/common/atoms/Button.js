import React from 'react';
import PropTypes from 'prop-types';
import { Text, View, TouchableOpacity, ActivityIndicator } from 'react-native';

const styles = {
  defaultTextStyle: {
    color: 'white',
    backgroundColor: 'rgba(0,0,0,0)',
    fontSize: 16,
  },
  defaultButtonStyle: {
    flexDirection: 'row',
    borderRadius: 2,
    alignItems: 'center',
    justifyContent: 'center',
    paddingHorizontal: 20,
    paddingVertical: 10,
    backgroundColor: 'rgba(0,0,0,0)',
  },
};

const Button = ({ title, onPress, textStyle, viewStyle, disabled, busy }) => {
  const { defaultButtonStyle, defaultTextStyle } = styles;

  const titleComponent =
    typeof title === 'string' ? <Text style={[defaultTextStyle, textStyle]}>{title}</Text> : title;

  return (
    <TouchableOpacity onPress={onPress} disabled={disabled}>
      <View style={[defaultButtonStyle, viewStyle, disabled && { opacity: 0.5 }]}>
        {busy && <ActivityIndicator size="small" style={{ marginRight: 4 }} />}
        {titleComponent}
      </View>
    </TouchableOpacity>
  );
};

Button.propTypes = {
  onPress: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  textStyle: Text.propTypes.style,
  viewStyle: Text.propTypes.style,
  disabled: PropTypes.bool,
  busy: PropTypes.bool,
};

Button.defaultProps = {
  textStyle: {},
  viewStyle: {},
  disabled: false,
  busy: false,
};

export default Button;
