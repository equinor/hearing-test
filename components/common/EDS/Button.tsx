import { Spinner } from "mad-expo-core";
import React from "react";
import { TouchableOpacity, View } from "react-native";

import Typography from "../atoms/Typography";

const ButtonEDS = (props: {
  onPress: () => void;
  text: string;
  outlined?: boolean;
  danger?: boolean;
  loading?: boolean;
  disabled?: boolean;
  style?: any;
}) => {
  function getBackgroundColor() {
    if (props.disabled || props.loading || props.outlined) return `#EAEAEA`;
    if (props.danger) return "#EB0000";
    return "#007079";
  }

  const { danger, disabled, loading, onPress, outlined, text } = props;
  return (
    <TouchableOpacity onPress={onPress} disabled={loading || disabled}>
      <View
        style={{
          height: 40,
          width: 160,
          alignSelf: "center",
          alignItems: "center",
          justifyContent: "center",
          borderRadius: 4,
          marginBottom: 18,
          borderWidth: !danger && 1,
          backgroundColor: getBackgroundColor(),
          borderColor: disabled || loading ? `#DCDCDC` : "#007079",
          ...props.style,
        }}
      >
        {loading ? (
          <Spinner />
        ) : (
          <Typography
            variant="button"
            color={outlined ? "#007079" : "white"}
            style={{
              textAlign: "center",
            }}
          >
            {text}
          </Typography>
        )}
      </View>
    </TouchableOpacity>
  );
};

ButtonEDS.defaultProps = {
  outlined: false,
  danger: false,
  loading: false,
  disabled: false,
};

export default ButtonEDS;
