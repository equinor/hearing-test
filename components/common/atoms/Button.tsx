import React from "react";
import {
  ActivityIndicator,
  GestureResponderEvent,
  StyleProp,
  StyleSheet,
  Text,
  TextStyle,
  TouchableOpacity,
  View,
  ViewStyle,
} from "react-native";

const Button = ({
  onPress,
  title,
  busy = false,
  disabled = false,
  textStyle = {},
  viewStyle = {},
}: {
  onPress: (event: GestureResponderEvent) => void;
  title: string | JSX.Element;
  busy?: boolean;
  disabled?: boolean;
  textStyle?: StyleProp<TextStyle>;
  viewStyle?: StyleProp<ViewStyle>;
}) => {
  const titleComponent =
    typeof title === "string" ? (
      <Text style={[styles.defaultTextStyle, textStyle]}>{title}</Text>
    ) : (
      title
    );

  return (
    <TouchableOpacity onPress={onPress} disabled={disabled}>
      <View
        style={[
          styles.defaultButtonStyle,
          viewStyle,
          disabled && { opacity: 0.5 },
        ]}
      >
        {busy && <ActivityIndicator size="small" style={{ marginRight: 4 }} />}
        {titleComponent}
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  defaultButtonStyle: {
    alignItems: "center",
    backgroundColor: "rgba(0,0,0,0)",
    borderRadius: 2,
    flexDirection: "row",
    justifyContent: "center",
    paddingHorizontal: 20,
    paddingVertical: 10,
  },
  defaultTextStyle: {
    backgroundColor: "rgba(0,0,0,0)",
    color: "white",
    fontSize: 16,
  },
});

export default Button;
