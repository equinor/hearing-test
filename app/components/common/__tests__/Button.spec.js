import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import Button from '../atoms/Button';

it('renders and behaves correctly', () => {
  const onButtonPress = sinon.spy();
  const wrapper = shallow(<Button title="Foo" onPress={onButtonPress} />);
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find('ActivityIndicator')).toHaveLength(0);
  expect(onButtonPress.calledOnce).toBeFalsy();
  wrapper.setProps({ busy: true });
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find('ActivityIndicator')).toHaveLength(1);
  wrapper.find('TouchableOpacity').simulate('press');
  expect(onButtonPress.calledOnce).toBeTruthy();
});
