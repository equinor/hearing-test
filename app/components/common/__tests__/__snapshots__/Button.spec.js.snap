// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders and behaves correctly 1`] = `ShallowWrapper {}`;

exports[`renders and behaves correctly 2`] = `ShallowWrapper {}`;
